hell/adventcalendar/2025/9/index.md

```diff
@@ -1,35 +1,273 @@
 
-## Heading
+To those of you new to building on the web, this is for you too, as you lucky things get to skip the old ways :)
 
-Some text.
+## Meet <button id="openMeetDialog">&lt;dialog&gt;</button>
 
+<dialog id="meetDialog" closedby="any">
+  <p>
+    Hello 👋
+    <br>Nice to meet you!
+  </p>
+  <form method="dialog">
+    <button>Close</button>
+  </form>
+</dialog>
 
-<p class="highlight"><strong>Note:</strong> Some text.</p>
+The [dialog element](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/dialog) has been available across browsers since March 2022, and allows you to pop a little overlay on your page - a common variant is known as a modal - and it has a whole bunch of abilities built in, with further useful facets still being added.
 
 ```html
-<h1> 
-  <a href="/">
-    Hello World
-  </a>
-</h1>
+<dialog id="meetDialog">
+  <p>
+    Hello 👋
+    <br />Nice to meet you!
+  </p>
+  <form method="dialog">
+    <button>Close</button>
+  </form>
+</dialog>
 ```
+
+## Modal and non-modal
+
+We're probably more familiar with the modal variety of dialog, intended to (hopefully only briefly) stop you interacting with the page under it. But there are also plenty of non-modal dialogs to be seen around the web, for example <button id="openToastDialog">toast</button> notifications.
+
+<dialog
+  id="toastDialog"
+  closedby="any"
+  style="
+    margin: 0;
+    border-width: 0.25em 1em 1em 0.25em;
+    border-style: solid;
+    border-radius: 1.5em 2em 0.5em 1em;
+    border-color: saddlebrown;
+    background-color: burlywood;
+    box-shadow: 1em 1em 1.5em -0.5em #8888;
+    rotate: 10deg;
+    position: fixed;
+    left: 3em;
+    bottom: 3em;
+    z-index: 10;
+  "
+>
+  <p>
+    Pop! Do I look
+    <br> a little burned?
+  </p>
+  <form method="dialog">
+    <button>Close</button>
+  </form>
+</dialog>
+
+Dialogs can give us both modal and non-modal variants, depending on how we open them. In the case of more than one dialog being opened (assuming no close action happening on the previous dialogs), the most recently opened dialog will appear on top.
+
+## Opening with JS
+
+To open a dialog, we use one of two JavaScript functions on a dialog element:
+
+```js
+.show()
+.showModal()
+```
+
+Likely based on the click of a button, you will run something like the following (you will likely use a much more specific `querySelector`):
+
+```js
+document.querySelector("dialog").show();
+// or
+document.querySelector("dialog").showModal();
+```
+
+<p class="highlight">💡
+  There is a declarative method available to <a href="">invoke</a> dialogs, but for now I will stick to what is currently available across the main browsers.
+</p>
+
+In the CodePen below, both kinds of dialog opening method are available. You'll notice that you can still interact with the Open Modal button after having opened the dialog, but not vice-versa. The modal can be closed with the Esc key, the normal modal cannot.
```

Comment:
I think with “normal modal” you mean the non modal dialog? Maybe say “non-modal dialog” instead of “normal modal”?